janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -14,37 +14,65 @@
+import org.apache.tinkerpop.gremlin.server.Settings;
+import org.janusgraph.diskstorage.configuration.backend.CommonsConfiguration;
 import org.janusgraph.graphdb.configuration.builder.GraphDatabaseConfigurationBuilder;
-import org.janusgraph.graphdb.management.JanusGraphManager;
+import org.janusgraph.graphdb.database.StandardJanusGraph;
 import org.janusgraph.graphdb.management.ConfigurationManagementGraph;
+import org.janusgraph.graphdb.management.JanusGraphManager;
 import org.janusgraph.graphdb.management.utils.ConfigurationManagementGraphNotEnabledException;
-import org.janusgraph.diskstorage.configuration.backend.CommonsConfiguration;
-import org.janusgraph.graphdb.database.StandardJanusGraph;
-import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.GRAPH_NAME;
-import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.STORAGE_BACKEND;
-
-import org.apache.tinkerpop.gremlin.server.Settings;
-import org.apache.commons.configuration.MapConfiguration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
 
-import java.util.Map;
 import java.util.HashMap;
+import java.util.Map;
 
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.AfterEach;
-import static org.junit.jupiter.api.Assertions.*;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.GRAPH_NAME;
+import static org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.STORAGE_BACKEND;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public abstract class ConfiguredGraphFactoryTest {
 
-public class ConfiguredGraphFactoryTest {
     private static final JanusGraphManager gm;
+
     static {
         gm = new JanusGraphManager(new Settings());
-        final Map<String, Object> map = new HashMap<>();
-        map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
-        final MapConfiguration config = new MapConfiguration(map);
+        final MapConfiguration config = getManagementConfig();
         final StandardJanusGraph graph = new StandardJanusGraph(new GraphDatabaseConfigurationBuilder().build(new CommonsConfiguration(config)));
         // Instantiate the ConfigurationManagementGraph Singleton
         new ConfigurationManagementGraph(graph);
     }
 
+    /**
+     * Getter for the settings to use to instantiate the graph supporting the
+     * ConfigurationManagementGraph instance.
+     * @return a MapConfiguration instance
+     */
+    protected static MapConfiguration getManagementConfig() {
+        final Map<String, Object> map = new HashMap<>();
+        map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
```

Comment:
Is it possible not to use in-memory backend in CQL test? Not familiar with this, so not sure if this is the reason you are unable to use JanusGraphCassandraContainer.

Reply:
This is a good idea, thanks.

I try it locally but `JanusGraphCassandraContainer` seems unreliable for me, I'm having a consistent `java.lang.IllegalStateException: Could not send request, session is closed` in the first call to `ConfiguredGraphFactory.removeConfiguration(graphName);`. I don't have this behavior with the Cassandra container.

I think this could be an issue with my local machine (I also had to restart docker sometimes for the container to start running) so I pushed this change to see if it works on the CI pipeline.